packages/flutter/lib/src/material/dialog.dart

```diff
@@ -481,10 +486,16 @@ class _DialogRoute<T> extends PopupRoute<T> {
 Future<T> showDialog<T>({
   @required BuildContext context,
   bool barrierDismissible: true,
-  @required Widget child,
+  @Deprecated(
+    'Instead of using the "child" argument, return the child from a closure '
+    'provided to the "builder" argument. This will ensure that the BuildContext'
+    ' is appropriate for widgets built in the dialog.'
```

Comment:
trivial nit: convention is to put the space on the end of the line rather than the start

Reply:
Done